types/src/main.ts

```diff
@@ -1470,6 +1479,39 @@ class TextEditChangeImpl implements TextEditChange {
 	}
 }
 
+/**
+ * An interactive text edit.
+ *
+ * @since 3.18.0
+ * @proposed
+ */
+export interface SnippetTextEdit {
+	/**
+	 * The range of the text document to be manipulated.
+	 */
+	range: Range;
+
+	/**
+	 * The snippet to be inserted.
+	 */
+	snippet: StringValue;
+
+	/**
+	 * The actual identifier of the snippet edit.
+	 */
+	annotationId?: ChangeAnnotationIdentifier;
+}
+
+export namespace SnippetTextEdit {
+	export function is(value: any): value is SnippetTextEdit {
+		const candidate = value as SnippetTextEdit;
+		return Is.objectLiteral(candidate)
+			&& Range.is(candidate.range)
+			&& StringValue.isSnippet(candidate.snippet)
+			&& (ChangeAnnotation.is(candidate.annotationId) || ChangeAnnotationIdentifier.is(candidate.annotationId));
```

Comment:
I haven't properly looked at this so apologies if I've misunderstood, but won't this check fail if `annotationId` is undefined? The interface above for `SnippetTextEdit ` suggests it's optional, but these checks seem to require that it's either a `ChangeAnnotation` or `ChangeAnnotationIdentifier`?

Reply:
You're right, good catch! #1408 should fix it.